src/layer.js

```diff
@@ -234,6 +234,32 @@ export class MapLayer extends HTMLElement {
       }
     }, 0);
   }
+  getOuterHTML() {
+    let fullLayer = this.outerHTML;
+    let tempElement = document.createElement('div');
+    tempElement.innerHTML = fullLayer;
```

Comment:
you can use `this.cloneNode(true)` and delete the cloned element at the end of the method, instead of doing this.